lib/src/edhoc.rs

```diff
@@ -151,48 +147,24 @@ pub fn r_prepare_message_2(
     ))
 }
 
-// FIXME fetch ID_CRED_I and CRED_I based on kid
 pub fn r_parse_message_3(
     state: &mut WaitM3,
     crypto: &mut impl CryptoTrait,
     message_3: &BufferMessage3,
-) -> Result<(ProcessingM3, CredentialRPK, Option<EADItem>), EDHOCError> {
+) -> Result<(ProcessingM3, IdCred, Option<EADItem>), EDHOCError> {
     let plaintext_3 = decrypt_message_3(crypto, &state.prk_3e2m, &state.th_3, message_3);
 
     if let Ok(plaintext_3) = plaintext_3 {
         let decoded_p3_res = decode_plaintext_3(&plaintext_3);
 
         if let Ok((id_cred_i, mac_3, ead_3)) = decoded_p3_res {
-            let id_cred_i = match id_cred_i {
-                IdCred::CompactKid(kid) => CredentialRPK {
-                    value: Default::default(),
-                    public_key: Default::default(),
-                    kid,
-                },
-                IdCred::FullCredential(cred) => {
-                    let Ok(buffer) = EdhocMessageBuffer::new_from_slice(cred) else {
-                        return Err(EDHOCError::ParsingError);
-                    };
-                    if let Ok(parsed_rpk) = CredentialRPK::new(buffer) {
-                        parsed_rpk
-                    } else {
-                        // This is incomplete, and the application will need to fill in the gaps --
-                        // just as in the CompactKid case the CredentialRPK is also incomplete.
-                        CredentialRPK {
-                            value: buffer,
-                            public_key: Default::default(),
-                            kid: Default::default(),
-                        }
-                    }
-                }
-            };
-
```

Comment:
Could you comment on why we don't need this block any more?

Reply:
According to [Section 5.4.3 of RFC 9528](https://www.rfc-editor.org/rfc/rfc9528.html#name-responder-processing-of-mess), ID_CRED_I is just made available to the application, which will process it. So even if it is a credential by value, we don't need to parse it here.

In addition, now the logic to parse ID_CRED_I is embedded in `decode_plaintext_3`, which calls `IdCred::from_encoded_value`. That's why we don't need to use different constructs for e.g. `CompactKid` or `FullCredential`.
Note that this only parses ID_CRED_I: from a (potentially compact-encoded) reference or value, e.g. `kid` / `{14: kccs}`, to the actual value of the ID_CRED_I, e.g.`{4: kid}` / `{14: kcss}`. Then, resolving the `kid` to a credential or verifying the syntax of the `kcss` is left to the application.

Reply:
Sounds good, thanks for the explanation!